torchinductor/dependencies.py

```diff
@@ -133,7 +133,9 @@ def canonicalize(self, index):
         # convert it to the simpliest form because of the interference from
         # different indexing formulas.
         index_vars = list(self._var_ranges.keys())
-        new_sizes, reindex, prune = _simplify_loops(index_vars, sizes, [index])
+        new_sizes, reindex, prune = _simplify_loops(
+            index_vars, sizes, [index], preserver_reoredering=True
```

Comment:
I think this line is dealing with the same issue:
https://github.com/pytorch/torchdynamo/blob/daa476aeb35d9cadd44f30b3c557584323fc25a0/torchinductor/codegen/triton.py#L581

That fix might better as if you have a,b,c it also prevents merging a+c.